airflow/providers/amazon/aws/hooks/glue_crawler.py

```diff
@@ -0,0 +1,216 @@
+        Checks if the crawler already exists
+
+        :param crawler_name: unique crawler name per AWS account
+        :type crawler_name: str
+        :return: Returns True if the crawler already exists and False if not.
+        """
+        self.log.info("Checking if AWS Glue crawler already exists: %s", crawler_name)
+
+        try:
+            self.glue_client.get_crawler(Name=crawler_name)
+            return True
+        except self.glue_client.exceptions.EntityNotFoundException:
+            return False
+
+    def update_crawler(self, **crawler_kwargs) -> str:
+        """
+        Updates crawler configurations
+
+        :param crawler_kwargs = Keyword args that define the configurations used for the crawler
+        :type crawler_kwargs = any
+        :return: True if crawler was updated and false otherwise
+        """
+        crawler_name = crawler_kwargs['Name']
+        current_crawler = self.glue_client.get_crawler(Name=crawler_name)['Crawler']
+
+        update_config = {
+            key: value for key, value in crawler_kwargs.items() if current_crawler[key] != crawler_kwargs[key]
+        }
+        if update_config != {}:
+            self.log.info("Updating crawler: %s", crawler_name)
+            self.glue_client.update_crawler(**crawler_kwargs)
+            self.log.info("Updated configurations: %s", update_config)
+            return True
+        else:
+            return False
+
+    def create_crawler(self, **crawler_kwargs) -> str:
+        """
+        Creates an AWS Glue Crawler
+
+        :param crawler_kwargs = Keyword args that define the configurations used to create the crawler
+        :type crawler_kwargs = any
+        :return: Name of the crawler
+        """
+        crawler_name = crawler_kwargs['Name']
+        self.log.info("Creating AWS Glue crawler: %s", crawler_name)
+
+        try:
+            glue_response = self.glue_client.create_crawler(**crawler_kwargs)
+            return glue_response['Crawler']['Name']
+        except self.glue_client.exceptions.InvalidInputException as general_error:
+            self.check_iam_role(crawler_kwargs['Role'])
+            raise AirflowException(general_error)
+
+    def start_crawler(self, crawler_name: str) -> dict:
+        """
+        Triggers the AWS Glue crawler
+
+        :param crawler_name: unique crawler name per AWS account
+        :type crawler_name: str
+        :return: Empty dictionary
+        """
+        crawler = self.glue_client.start_crawler(Name=crawler_name)
+        return crawler
+
+    def get_crawler_state(self, crawler_name: str) -> str:
+        """
+        Get state of the Glue crawler. The crawler state can be
+        ready, running, or stopping.
+
+        :param crawler_name: unique crawler name per AWS account
+        :type crawler_name: str
+        :return: State of the Glue crawler
+        """
+        crawler = self.glue_client.get_crawler(Name=crawler_name)
+        crawler_state = crawler['Crawler']['State']
```

Comment:
Are we sure we need separate methods for one field? Can't we generalize it more?  During one iteration of the loop within the `wait_for_crawler_completion ` function, the crawler is fetched 3 times. It's a bit of a lot.

Reply:
maybe consolidate `get_crawler_status()` and `get_crawler_state()` into one `get_crawler()` function that returns the results of the `glue_client.get_crawler()` and then just index into the response to get the state/status within `wait_for_crawler_completion()`?

something like this:

```
while True:
            crawler = self.get_crawler(crawler_name)
            crawler_state = crawler['Crawler']['State']
            if crawler_state == 'READY':
                self.log.info("State: %s", crawler_state)
                crawler_status = crawler['Crawler']['LastCrawl']['Status']
                if crawler_status in failed_status:
                    raise AirflowException(
                        f"Status: {crawler_status}"
                    )  # pylint: disable=raising-format-tuple
                else:
                    metrics = self.get_crawler_metrics(crawler_name)
                    self.log.info("Status: %s", crawler_status)
                    self.log.info("Last Runtime Duration (seconds): %s", metrics['LastRuntimeSeconds'])
                    self.log.info("Median Runtime Duration (seconds): %s", metrics['MedianRuntimeSeconds'])
                    self.log.info("Tables Created: %s", metrics['TablesCreated'])
                    self.log.info("Tables Updated: %s", metrics['TablesUpdated'])
                    self.log.info("Tables Deleted: %s", metrics['TablesDeleted'])

                    return crawler_status
```